src/Diagnostics/FxCop/SDL.Analyzer.Common/CSharp/Shared/SyntaxNodeHelper.cs

```diff
@@ -0,0 +1,323 @@
+                SyntaxKind kind = node.Kind();
+                if ((kind == SyntaxKind.InvocationExpression) && ((callKind & CallKind.Invocation) != 0))
+                {
+                    var invocationNode = (InvocationExpressionSyntax)node;
+                    argList = invocationNode.ArgumentList;
+                }
+                else if ((kind == SyntaxKind.ObjectCreationExpression) && ((callKind & CallKind.ObjectCreation) != 0))
+                {
+                    var invocationNode = (ObjectCreationExpressionSyntax)node;
+                    argList = invocationNode.ArgumentList;
+                }
+                if (argList != null)
+                {
+                    return argList.Arguments.Select(arg => arg.Expression);
+                }
+            }
+
+            return Enumerable.Empty<SyntaxNode>();
+        }
+
+        public override IEnumerable<SyntaxNode> GetObjectInitializerExpressionNodes(SyntaxNode node)
+        {
+            var empty = Enumerable.Empty<SyntaxNode>();
+            if (node == null)
+            {
+                return empty;
+            }
+
+            SyntaxKind kind = node.Kind();
+            if (kind != SyntaxKind.ObjectCreationExpression)
+            {
+                return empty;
+            }
+
+            var objectCreationNode = (ObjectCreationExpressionSyntax)node;
+            if (objectCreationNode.Initializer == null)
+            {
+                return empty;
+            }
+
+            return objectCreationNode.Initializer.Expressions;
+        }
+
+        public override bool IsMethodInvocationNode(SyntaxNode node)
+        { 
+            if (node == null)
+            {
+                return false;
+            }
+            SyntaxKind kind = node.Kind();
+            return kind == SyntaxKind.InvocationExpression || kind == SyntaxKind.ObjectCreationExpression;
+        }
+
+        public override bool IsObjectCreationExpressionUnderFieldDeclaration(SyntaxNode node)
+        {
+            return node != null &&
+                   node.Kind() == SyntaxKind.ObjectCreationExpression &&
+                   node.AncestorsAndSelf().OfType<FieldDeclarationSyntax>().FirstOrDefault() != null;
+        }
+
+        public override SyntaxNode GetVariableDeclaratorOfAFieldDeclarationNode(SyntaxNode node)
+        {
+            if (IsObjectCreationExpressionUnderFieldDeclaration(node))
+            {
+                return node.AncestorsAndSelf().OfType<VariableDeclaratorSyntax>().FirstOrDefault();
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        public override IEnumerable<SyntaxNode> GetDescendantAssignmentExpressionNodes(SyntaxNode node)
+        {
+            var empty = Enumerable.Empty<SyntaxNode>();
+            if (node == null)
+            {
+                return empty;
+            }
+
+            return node.DescendantNodesAndSelf().OfType<AssignmentExpressionSyntax>();
```

Comment:
I think we need to clean up all helpers to avoid traversing descendants or ancestors. This will hurt the analyzer performance when executed for all code blocks for all nodes contained in it!